opencog/util/Logger.cc

```diff
@@ -147,26 +148,33 @@ Logger::~Logger()
     flush();
     stop_write_loop();
 
-    if (logfile != NULL) fclose(logfile);
+    {
+        std::unique_lock<std::mutex> lock(the_mutex);
+        if (logfile != NULL) fclose(logfile);
+    }
 }
 
 void Logger::start_write_loop()
 {
-    std::unique_lock<std::mutex> lock(the_mutex);
-    if (!writingLoopActive)
     {
-        writingLoopActive = true;
-        writer_thread = std::thread(&Logger::writing_loop, this);
+        std::unique_lock<std::mutex> lock(the_mutex);
```

Comment:
vsbogd 2 minutes ago • Member

It looks I have misleaded you a bit. If all method body should be protected by the mutex it is not necessary to create additional code block. So here it should be just as ut was:

void Logger::start_write_loop()
{
    std::unique_lock<std::mutex> lock(the_mutex);
    if (!writingLoopActive)
    {
        writingLoopActive = true;
        writer_thread = std::thread(&Logger::writing_loop, this);
    }
}